File: titanic/titanic.py
#Primer par de graficos

plt.figure(figsize=(15, 4), dpi=70)
plt.subplot(1,2,1)
plt.title("Survived vs Pclass")
plt.xlabel("Pclass")
plt.ylabel("Survived")

plt.hist([df[df['Survived']>0]['Pclass'],df[df['Survived']==0]['Pclass']], stacked=True, color = ['g','r'],
         bins = 30,label = ['Not survive','Survived'])

plt.subplot(1,2,2)
plt.title("Survived vs Sex")
plt.xlabel("Sex")
plt.ylabel("Survived")

plt.hist([df[df['Survived']>0]['Sex'],df[df['Survived']==0]['Sex']], stacked=True, color = ['g','r'],
         bins = 30,label = ['Not survive','Survived'])
plt.show()

#Segundo par de graficos

plt.figure(figsize=(15, 4), dpi=70)
plt.subplot(1,2,1)
plt.title("Survived vs Age")
plt.xlabel("Age")
plt.ylabel("Survived")

plt.hist([df[df['Survived']>0]['Age'],df[df['Survived']==0]['Age']], stacked=True, color = ['g','r'],
         bins = 30,label = ['Not survive','Survived'])

plt.subplot(1,2,2)
plt.title("Survived vs SibSp")
plt.xlabel("SibSp")
plt.ylabel("Survived")

plt.hist([df[df['Survived']>0]['SibSp'],df[df['Survived']==0]['SibSp']], stacked=True, color = ['g','r'],
         bins = 30,label = ['Not survive','Survived'])
plt.show()

#Tercer par de graficos

plt.figure(figsize=(15, 4), dpi=70)
plt.subplot(1,2,1)
plt.title("Survived vs Parch")
plt.xlabel("Parch")
plt.ylabel("Survived")

plt.hist([df[df['Survived']>0]['Parch'],df[df['Survived']==0]['Parch']], stacked=True, color = ['g','r'],
         bins = 30,label = ['Not survive','Survived'])

plt.subplot(1,2,2)
plt.title("Survived vs Fare")
plt.xlabel("Fare")
plt.ylabel("Survived")

plt.hist([df[df['Survived']>0]['Fare'],df[df['Survived']==0]['Fare']], stacked=True, color = ['g','r'],
         bins = 30,label = ['Not survive','Survived'])
plt.show()

#Cuarto par de graficos

plt.figure(figsize=(15, 4), dpi=70)
plt.subplot(1,2,1)
plt.title("Survived vs Embarked")
plt.xlabel("Embarked")
plt.ylabel("Survived")

plt.hist([df[df['Survived']>0]['Embarked'],df[df['Survived']==0]['Embarked']], stacked=True, color = ['g','r'],
         bins = 30,label = ['Not survive','Survived'])

plt.subplot(1,2,2)
plt.show()

File: titanic/R/titanic.R
library(ggplot2) 
library(plotly)

data_titanic <- read.csv("train.csv", header = TRUE, sep = ",", quote = "\"", dec = ".", fill = TRUE, comment.char = "")

View(head(data_titanic))

df <- as.data.frame(data_titanic)

df$Name <- NULL
df$Cabin <- NULL
df$Ticket <- NULL
df$PassengerId <- NULL

View(df)

plot(data_titanic$Survived,data_titanic$PassengerId)

hist(df$Pclass[df$Survived==0])
hist(df$Pclass[df$Survived==1])

sum(is.na(df$Survived))
sum(is.na(df$Pclass))
sum(is.na(df$Sex))
sum(is.na(df$Age))
sum(is.na(df$SibSp))
sum(is.na(df$Parch))
sum(is.na(df$Fare))
sum(is.null(df$Embarked))

df$Sex<-as.numeric(df$Sex)
df$Embarked<-as.numeric(df$Embarked)

View(df)
